sklearn/utils/tests/test_sparsefuncs.py

```diff
@@ -478,3 +480,10 @@ def test_csc_row_median():
 
     # Test that it raises an Error for non-csc matrices.
     assert_raises(TypeError, csc_median_axis_0, sp.csr_matrix(X))
+
+
+def test_inplace_normalize():
+    # regression tests for passing 32-bit floating point
+    X = sp.rand(10, 5, dtype=np.float32, density=.5).tocsr()
+    inplace_csr_row_normalize_l1(X)
+    inplace_csr_row_normalize_l2(X)
```

Comment:
This does not check that X has been properly normalized, and in practice it cannot be the case as the data is array which is actually normalized is a copy of the input `X.data` when the dtype is not `np.float64`.